src/shogun/base/AnyParameter.h

```diff
@@ -220,6 +229,11 @@ namespace shogun
 			return m_init_function;
 		}
 
+		std::function<bool(Any)> get_constrain_function() const
```

Comment:
should this return `std::function<bool(Any)>&`? I am not sure if this is making a copy of the function each time?

Reply:
quick check on compiler explorer and it seems like in O1 returning by reference produces fewer instructions!